js/framework/AnalyticsConstants.js

```diff
@@ -9,37 +9,356 @@ if (!OO.Analytics.EVENTS)
+     * OO.Analytics.EVENT_DATA.VideoSeekStartedData.
+     */
+    VIDEO_SEEK_STARTED:             'video_seek_started',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_SEEK_ENDED
+     * @description This message is sent when a video seek has completed.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoSeekEndedData.
+     */
+    VIDEO_SEEK_ENDED:               'video_seek_ended',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_STREAM_DOWNLOADING
+     * @description This message is sent when a video stream is downloading data
+     * but still playing video. If the stream has to stop and load data, that is
+     * considered a buffering event.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoDownloadingMetadata.
+     */
     VIDEO_STREAM_DOWNLOADING:       'video_stream_downloading',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_BUFFERING_STARTED
+     * @description This message is sent when a video stream has to pause playback
+     * to load more data.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoBufferingStartedData.
+     */
     VIDEO_BUFFERING_STARTED:        'video_buffering_started',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#VIDEO_BUFFERING_ENDED
+     * @description This message is sent when a video stream has buffering and
+     * is ready to resume playback.
+     * @param {Array} paramArray Array of length 1, contains an instance of
+     * OO.Analytics.EVENT_DATA.VideoBufferingEndedData.
+     */
     VIDEO_BUFFERING_ENDED:          'video_buffering_ended',
 
+    /**
+     * @public
+     * @event Analytics#EVENTS#AD_BREAK_STARTED
+     * @description This message is sent when the player stops the main content
+     * to start playing ads.
+     */
     AD_BREAK_STARTED:               'ad_break_started',
+
+    /**
+     * @public
+     * @event Analytics#EVENTS#AD_BREAK_ENDED
+     * @description This message is sent when the player has finished playing ads
+     * and is ready to playback the main video.
+     */
     AD_BREAK_ENDED:                 'ad_break_ended',
 
+    /**
+     * @public
+     * @event Analytics#EVENTS#DESTROY
+     * @description This message is sent when the player it's plugins are destroying.
+     */
     DESTROY:                        'destroy'
   };
   OO.Analytics.EVENTS = EVENTS;
 }
 
+if (!OO.Analytics.EVENT_DATA)
+{
+  const EVENT_DATA = {};
+
+  /**
+   * @public
+   * @class VideoSourceData
+   * @classdesc Contains the data passed along with VIDEO_SOURCE_CHANGED. This
+   * includes the embed code (video id) and any metadata this video stream needs
+   * pass along to other plugins (Ex. It could contain ad tag data or analytics
+   * account information).
```

Comment:
I don't think you need to say what it includes in the description since it's in the properties